components/button.css

```diff
@@ -0,0 +1,173 @@
+/* buttons and links start here */
+
+/* links */
+
+/* Primary link */
+
+.link {
+  font-size: inherit;
+  color: var(--primary);
+  font-weight: bold;
+  text-decoration: underline;
+  text-underline-position: under;
+}
+.link:hover {
+  color: var(--secondary);
+}
+
+/* secondary link */
+
+.link-secondary {
+  font-size: inherit;
+  color: var(--font-lighter);
+  text-decoration: underline;
+  text-underline-position: under;
+  font-weight: bold;
+}
+.link-secondary:hover {
+  color: var(--font-light);
+}
+
+/* Buttons */
+
+/* button with size */
+
+.btn-large {
+  font-weight: 700;
+  cursor: pointer;
+  font-size: 2.1rem;
+  width: 121px;
+  height: 48px;
+  padding: 8px 16px;
+  min-width: fit-content;
+  min-height: fit-content;
+  border: none;
+  border-radius: 15px;
+}
+.btn-medium {
+  font-weight: 700;
+  width: 100px;
+  height: 40px;
+  cursor: pointer;
+  font-size: 1.6rem;
+  background-color: var(--primary);
+  padding: 8px 16px;
+  min-width: fit-content;
+  min-height: fit-content;
+  border: none;
+  border-radius: 10px;
+  color: var(--white);
+}
+.btn-small {
+  font-weight: 700;
+  cursor: pointer;
+  font-size: 1.4rem;
+  background-color: var(--primary);
+  width: 91px;
+  height: 32px;
+  padding: 8px 16px;
+  min-width: fit-content;
+  min-height: fit-content;
+  border: none;
+  border-radius: 5px;
+  color: var(--white);
+}
+.btn-rounded {
+  border-radius: 30px;
+}
+
+/* button with icons */
+
+.btn-w-icon :nth-child(1) {
+  margin-right: 8px;
+}
+
+/* icon button */
+
+.btn-icon {
+  display: inline-flex;
+  justify-content: center;
+  align-items: center;
+  color: var(--primary);
+  font-size: var(--s);
+  padding: 8px;
+  width: min-content;
+  height: min-content;
+  border-radius: 50%;
+  border: none;
+  cursor: pointer;
+}
+
+/* button float */
+
+.btn-float {
+  display: inline-flex;
+  justify-content: center;
+  align-items: center;
+  text-decoration: none;
+  color: var(--white);
+  font-size: 1.6rem;
+  width: 50px;
+  height: 50px;
+  border-radius: 50%;
+  background-color: var(--primary);
+}
+.btn-float:hover {
+  background-color: var(--hover);
+}
+
+/* primary button */
+
+.btn-primary {
+  color: var(--white);
+  background-color: var(--primary);
+}
+.btn-primary:hover {
+  background-color: var(--hover);
+}
+
+/* button secondary */
+
+.btn-secondary {
+  background-color: transparent;
+  border: solid 2px var(--primary);
+  color: var(--primary);
+}
+.btn-secondary:hover {
+  border: solid 2px var(--hover);
+  color: var(--hover);
+}
+
+/* button disabled */
+
+.btn-disabled {
+  background-color: var(--font-lighter);
+  cursor: auto;
+}
+.btn-disabled:hover {
+  background-color: var(--font-lighter);
+  cursor: auto;
+}
+
+/* close button */
+
+.btn-close {
+  background-color: transparent;
+  border: none;
+  cursor: pointer;
+  color: inherit;
+}
+
+/* button vertical */
+
+.btn-vertical :nth-child(n) {
```

Comment:
I am skeptic if using :nth-child is a good practice. Could you go for more specific selectors. Like, if you want to select all the children of an element, you can use '> *' selector.

Reply:
Yes I will make the changes